internal/remote/client_factory.go

```diff
@@ -0,0 +1,107 @@
+package remote
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	apicorev1 "k8s.io/api/core/v1"
+	apimetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8slabels "k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+)
+
+type SkrContextFactory interface {
+	Get(kyma types.NamespacedName) (*SkrContext, error)
+	Init(ctx context.Context, kyma types.NamespacedName) error
```

Comment:
I think we can also introduce `DeleteCache(kyma types.NamespacedName)` to this interface, and then we can get rid of that `RemoteClientCache` in `KymaReconciler struct`